search.go

```diff
@@ -27,80 +27,142 @@ import (
 // ErrNoQuery is returned when an empty query is given
 var ErrNoQuery = errors.New("no query given")
 
-type Query interface {
-	// And adds a condition to query on
-	And(part QueryPart) Query
+type jsonPath string
 
-	// Parts returns the different parts of the query
-	Parts() []QueryPart
+// NewJSONPath creates a JSON path query: "person.path" or "person.children.#.path"
+// # is used to traverse arrays
+func NewJSONPath(path string) QueryPath {
+	return jsonPath(path)
 }
 
-type QueryPart interface {
+func (q jsonPath) Equals(other QueryPath) bool {
+	otherJSONPath, ok := other.(jsonPath)
```

Comment:
is this cast needed, if you just use the equality operator below?

Reply:
it's not